WikiWatershed/model-my-watershed
# Add other NLCD layers to Land Cover Modifications

## Overview

Adds all NLCD layers to the Land Cover Modification dialog in MapShed projects. The default layer is selected based on the project's override.

Connects #3409

### Demo

![image](https://user-images.githubusercontent.com/1430060/127579936-243ba8d3-5e73-4921-8cd3-5dbb45045a11.png)

### Notes

In 1d297d9 and 13a4202 functionality is added to disable land cover options whose analysis is not complete yet, so they are not accidentally selected by the user. But this list does not update as the analysis completes. Instead, the dialog must be closed and re-opened for the new status to take effect.

To update the list as the analyses complete was beyond the scope of this card, and will be deferred until the future: #3413.

## Testing Instructions

* Check out this branch and `./scripts/bundle.sh --debug`
* Create a new MapShed project and add a scenario to it
* Open the Land Cover Modification dialog
  - [x] Ensure you see all NLCD layers in the Preset dropdown
  - [x] Ensure the correct layer is set as the default
  - [x] Ensure selecting a preset fills in all the land cover values as expected
